homeassistant/components/accuweather/sensor.py

```diff
@@ -413,34 +392,32 @@ def __init__(
             self._attr_unique_id = (
                 f"{coordinator.location_key}-{description.key}".lower()
             )
-        self._attr_native_unit_of_measurement = description.native_unit_of_measurement
-        if self.coordinator.hass.config.units is METRIC_SYSTEM:
-            self._unit_system = API_METRIC
-            if metric_unit := description.metric_unit:
-                self._attr_native_unit_of_measurement = metric_unit
-        else:
-            self._unit_system = API_IMPERIAL
-            if us_customary_unit := description.us_customary_unit:
-                self._attr_native_unit_of_measurement = us_customary_unit
         self._attr_device_info = coordinator.device_info
-        if forecast_day is not None:
-            self.forecast_day = forecast_day
+        self.forecast_day = forecast_day
+        if (
+            description.device_class == SensorDeviceClass.WIND_SPEED
+            and coordinator.hass.config.units == US_CUSTOMARY_SYSTEM
+        ):
+            self._attr_suggested_unit_of_measurement = UnitOfSpeed.MILES_PER_HOUR
```

Comment:
I think this is needed due to a mistake in the unit system, correction here: https://github.com/home-assistant/core/pull/90504

Reply:
Awesome, I'll wait for that.